spec/classes/init_spec.rb

```diff
@@ -950,22 +950,29 @@
     end
   end
 
-  context 'with sshd_config_port not being a valid number' do
+  describe 'sshd_config_port param' do
     let :facts do
-      default_facts.merge(
+      default_facts_merge(
         {
+          :root_home => '/root',
         }
       )
     end
-    let :params do
-      { :sshd_config_port => '22invalid' }
-    end
 
-    it 'should fail' do
-      expect {
-        should contain_class('ssh')
-      }.to raise_error(Puppet::Error,/ssh::sshd_config_port must be a valid number and is set to <22invalid>\./)
-    end
+  context 'when set to an array' do
+    let (:params) {{'sshd_config_port' => ['22222', '22223'] }}
+
+    it { should contain_file('sshd_config').with_content(/^Port 22222\nPort 22223$/) }
+  end
+
+  context 'when set to a string' do
+    let (:params) {{'sshd_config_port' => ['22222'] }}
```

Comment:
Those two tests are good, though they need renamed a bit.

Needs an additional test where sshd_config_port is a string. A string looks like this. Notice no square brackets.

``` ruby
let (:params) {{'sshd_config_port' => '22222' }}
```